stock_split_picking/models/stock_picking.py

```diff
@@ -12,34 +12,38 @@ class StockPicking(models.Model):
 
     _inherit = "stock.picking"
 
-    def _check_split_process(self):
+    def _check_split_process(self, split_move_type):
         # Check the picking state and condition before split
         if self.state == "draft":
             raise UserError(_("Mark as todo this picking please."))
-        if all([x.qty_done == 0.0 for x in self.move_line_ids]):
-            raise UserError(
-                _(
-                    "You must enter done quantity in order to split your "
-                    "picking in several ones."
+
+        if split_move_type == "quantity_done":
+            if all([x.qty_done == 0.0 for x in self.move_line_ids]):
+                raise UserError(
+                    _(
+                        "You must enter done quantity in order to split your "
+                        "picking in several ones."
+                    )
                 )
-            )
 
-    def split_process(self):
+    def split_process(self, split_move_type):
```

Comment:
As this parameter is in fact a field of stock move, maybe its name should reflect that ?

Reply:
A docstring should be great too